src/main/java/dk/kb/labsapi/ImageExport.java

```diff
@@ -0,0 +1,319 @@
+        startYear = conf.getInteger(".imageserver.minYear");
+        endYear = conf.getInteger(".imageserver.maxYear");
+        log.info("Created ImageExport that exports images from this server: '{}'", ImageExportService);
+    }
+
+    /**
+     * Get link to image from newspaper page with given query present in text.
+     * @param query     to search for.
+     * @param startTime is the earliest boundary for the query.
+     * @param endTime   is the latest boundary for the query.
+     * @param max       number of documents to query for illustrations.
+     * @param output    to write images to as one combined zip file.
+     */
+    public void getImageFromTextQueryAsStream(String query, Integer startTime, Integer endTime, Integer max, OutputStream output) throws IOException {
+        if (instance.ImageExportService == null) {
+            throw new InternalServiceException("Illustration delivery service has not been configured, sorry");
+        }
+        // Query Solr
+        QueryResponse response = illustrationSolrCall(query, startTime, endTime, max);
+        // Get illustration metadata
+        List<IllustrationMetadata> illustrationMetadata = getMetadataForIllustrations(response);
+        // Get illustration URLS
+        List<URL> illustrationUrls = createLinkForAllIllustrations(illustrationMetadata);
+        // Streams illustration from URL to zip file with all illustrations
+        illustrationURLSToStream(illustrationUrls, illustrationMetadata, output);
+    }
+
+    /**
+     * Call Solr for input query and return fields needed to extract images from pages in the query.
+     * @param query     to query Solr with.
+     * @param startTime startTime is the earliest boundary for the query.
+     * @param endTime   endTime is the latest boundary for the query.
+     * @param max       number of results to return
+     * @return a response containing specific metadata used to locate illustration on pages. The fields returned are the following: <em>pageUUID, illustration, page_width, page_height</em>
+     */
+     public QueryResponse illustrationSolrCall(String query, Integer startTime, Integer endTime, int max) throws IOException{
+         // Check start and end times
+         int usableStartTime = setUsableStartTime(startTime);
+         int usableEndTime = setUsableEndTime(endTime);
+         log.info("Usable start time is: " + usableStartTime);
+         log.info("usable end time is: " + usableEndTime);
+         if (usableStartTime > usableEndTime){
+             log.error("The variable startTime is greater than endTime, which is not allowed. Please make startTime less than endTime.");
+             throw new IOException("The variable startTime is greater than endTime, which is not allowed. Please make startTime less than endTime.");
+         }
+
+        // Construct solr query with filter
+        String filter = "recordBase:doms_aviser AND py:[" + usableStartTime + " TO "+ usableEndTime + "]";
```

Comment:
I really need to fix filter-thing this so we don't have to rely on a steady hand to ensure people only have access to a specific date range.